docs/rollup-boost-ha.md

```diff
@@ -0,0 +1,245 @@
+
+- **Payload Production**:  
+  During each `get_payload` request, Rollup Boost will verify payload availability from both the builder and the execution client. If the builder fails to deliver a payload, Rollup Boost will report partial health. If the execution client fails to deliver a payload, Rollup Boost will report unhealthy.
+
+`op-conductor` should also be configurable in how it interprets health status for failover decisions. This allows chain operators to define thresholds based on their risk tolerance and operational goals. For example, operators may choose to maintain leadership with a sequencer reporting `206 Partial Content` to avoid unnecessary fail overs or they may configure `op-conductor` to immediately fail over when any degradation is detected. This flexibility allows the chain operator to configure a failover policy that aligns with network performance expectations and builder reliability.
+
+<br>
+
+| Condition | Health Status |
+|:----------|:--------------|
+| Builder is synced and both execution client and builder return payloads | `200 OK` (Healthy) |
+| Builder is out of sync| `206 Partial Content` (Partially Healthy) |
+| Builder fails to return payload on `get_payload` request | `206 Partial Content` (Partially Healthy) |
+| Execution client fails to return payload on `get_payload` request | `503 Service Unavailable` (Unhealthy) |
+
+`op-conductor` should query the `/healthz` endpoint exposed by Rollup Boost in addition to the existing execution client health checks. Health should be interpreted as follows:
+
+- `200 OK` (Healthy): The node is fully healthy and eligible for leadership.
+- `206 Partial Content` (Partially Healthy): The node is degraded but may be considered for leadership if configured by operator
+- `503 Service Unavailable` (Unhealthy): The node is unhealthy and must be excluded from leadership.
+
+During normal operation and leadership transfers, `op-conductor` should prioritize candidates in the following order:
+
+1. Prefer nodes reporting `200 OK`.
+2. If no fully healthy nodes are available, select from nodes reporting `206 Partial Content`.
+3. Nodes reporting `503 Service Unavailable` must not be selected as leader.
+
+Rollup Boost instances that are not actively sequencing rely exclusively on the builder sync check to report health, as they are not producing blocks. This behavior mirrors the existing `op-conductor` health checks for inactive sequencers and ensures readiness during failover without compromising network liveness guarantees.
```

Comment:
Can you please expand on that? Why don't we want to have follower node report local sequencer health?

For example, if the follower got elected leader, builder is in sync, but the local sequencer is out of sync, we'd want to transfer leader to another healthy one, right?

Reply:
> Can you please expand on that? Why don't we want to have follower node report local sequencer health?

For clarity, this section isn’t proposing that `op-conductor` should rely solely on the builder health checks. The existing sequencer health checks would still be used to evaluate sequencer status as they are today. This is additive, what’s described here introduces additional health checks focused specifically on assessing builder health alongside the existing ones. Follower nodes would still report local sequencer health status.

Reply:
I think we're aligned, but should we update the message to remove `exclusively`? Maybe change to `additionally`?

Reply:
Sounds good, I have updated this section to reflect this. Let me know if you think it should be adjusted further.

> Rollup Boost instances that are not actively sequencing rely on the builder sync check to report health, as they are not producing blocks.  This behavior mirrors the existing `op-conductor` health checks for inactive sequencers and ensures readiness during failover without compromising network liveness guarantees. Note that `op-conductor` will still evaluate existing sequencer health checks to determine overall sequencer health.